pkg/process/runner/submitter.go

```diff
@@ -212,7 +213,12 @@ func (s *CheckSubmitter) Start() error {
 		defer s.wg.Done()
 
 		heartbeat := time.NewTicker(15 * time.Second)
-		defer heartbeat.Stop()
+		// We only want to send heartbeats for the process agent process
+		if flavor.GetFlavor() == flavor.ProcessAgent {
+			defer heartbeat.Stop()
+		} else {
+			heartbeat.Stop()
```

Comment:
Is there a problem with not starting it, it seems to be hacky to create it and stop right away. Is `Codecov` problem flagged above real?

Reply:
As far as I could tell, there doesn't seem to be another good alternative. I had considered having a nil ticker that only gets set when in the process agent, but select cases don't allow for conditional checks, only sending or receiving on a channel. To get around that, you could create another sort of "wrapper" channel, but that doesn't seem very functionally different. As for the codecov, thats probably true, but I am currently trying to add a test.

Reply:
I actually was able to come up with a maybe less hacky solution by using a separate goroutine and added the test. Let me know what you think.